### PR TITLE
[receivertest] add metrics support to contract_checker

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
add support for metrics in contract checker

### CODE REVIEW

File: receiver/receivertest/contract_checker.go [hunk 1]
```diff
@@ -390,7 +390,76 @@ func idSetFromLogs(data plog.Logs) (idSet, error) {
...
 }
 
-// TODO: Implement mockConsumer.ConsumeMetrics()
+func (m *mockConsumer) ConsumeMetrics(_ context.Context, data pmetric.Metrics) error {
+	ids, err := idSetFromMetrics(data)
+	require.NoError(m.t, err)
+	return m.consume(ids)
+}
+
+// idSetFromLogs computes an idSet from given plog.Logs. The idSet will contain ids of all log records.
```
Comment (dmitryax) [inline receiver/receivertest/contract_checker.go:399]:
```suggestion
// idSetFromLogs computes an idSet from given pmetric.Metrics. The idSet will contain ids of all metric data points.
```

File: receiver/receivertest/contract_checker.go [hunk 2]
```diff
@@ -437,3 +506,42 @@ func CreateOneLogWithID(id UniqueIDAttrVal) plog.Logs {
 	)
 	return data
 }
+
+func CreateEveryMetricTypeWithID(id UniqueIDAttrVal) pmetric.Metrics {
```
Comment (dmitryax) [inline receiver/receivertest/contract_checker.go:510]: I don't think we should use the same id across several metrics. The ID is supposed to be unique per datapoint

Comment (atoulme) [reply receiver/receivertest/contract_checker.go]: alright, changed

File: receiver/receivertest/contract_checker_test.go [hunk 3]
```diff
@@ -52,30 +97,108 @@ func (s *exampleReceiver) Receive(data plog.Logs) {
...
+	data := CreateOneSpanWithID(id)
+
+	// Send the generated data to the receiver.
+	g.receiver.ReceiveTraces(data)
+
+	// And return the ids for bookkeeping by the test.
+	return []UniqueIDAttrVal{id}
+}
+
+func (g *exampleLogGenerator) GenerateTraces() []UniqueIDAttrVal {
```
Comment (dmitryax) [inline receiver/receivertest/contract_checker_test.go:155]: Why is this method added?

Comment (atoulme) [reply receiver/receivertest/contract_checker_test.go:155]: removed

File: receiver/receivertest/contract_checker.go (hunk 2 above)
Comment (dmitryax) [reply receiver/receivertest/contract_checker.go:510]: This isn’t addressed. Why do we need every metric type with the same ID? Can this method take metric type instead?

Comment (atoulme) [reply receiver/receivertest/contract_checker.go]: Sorry, bad push. Should be good now.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**